src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs

```diff
@@ -121,11 +123,17 @@ CollectionExpressionSyntax CreateCollectionExpressionWithoutExistingElements()
                 using var _ = ArrayBuilder<SyntaxNodeOrToken>.GetInstance(out var nodesAndTokens);
                 CreateAndAddElements(matches, nodesAndTokens, preferredIndentation: elementIndentation, forceTrailingComma: true);
 
+                var closeBracket = Token(SyntaxKind.CloseBracketToken).WithLeadingTrivia(Whitespace(openBraceIndentation));
+
+                // Add a newline between the last element and the close bracket if we don't already have one.
+                if (nodesAndTokens.Count > 0 && nodesAndTokens.Last().GetTrailingTrivia() is [.., (kind: not SyntaxKind.EndOfLineTrivia)])
+                    closeBracket = closeBracket.WithPrependedLeadingTrivia(endOfLine);
```

Comment:
📝 This is a bit of a tricky point. Technically the EndOfLineTrivia here should be appended to the trailing trivia of the preceding item, as opposed to prepending it to the leading trivia of the `]`. For my changes in #69309, I found edge cases where this distinction mattered for the behavior.

Reply:
fair.  let me tweak.

Reply:
done.